### ISSUE TITLE
Scheduler doesn't work with custom Runner

### ISSUE THREAD

**Issue by MartSlaaf** [catalyst-team/catalyst#1064]
## 🐛 Bug Report
When creating a custom runner as described in the documentation, the scheduler parameter is accepted, but never called.

### How To Reproduce
Steps to reproduce:
1. Create any sort of custom runner from the `Runner` parent class.
2. Pass scheduler as a parameter to `.train` call.

#### Code sample
```python
import torch, torchvision
from torch import nn

from torchvision.datasets import MNIST
from torch.utils.data import DataLoader

from catalyst.dl.runner import SupervisedRunner, Runner

train_loader = DataLoader(MNIST('mnist_test', train=True, download=True,
                             transform=torchvision.transforms.Compose([
                               torchvision.transforms.ToTensor(),
                               torchvision.transforms.Normalize(
                                 (0.1307,), (0.3081,))
                             ])),
  batch_size=128, shuffle=True)

test_loader = DataLoader(MNIST('mnist_test', train=False, download=True,
                             transform=torchvision.transforms.Compose([
                               torchvision.transforms.ToTensor(),
                               torchvision.transforms.Normalize(
                                 (0.1307,), (0.3081,))
                             ])),
  batch_size=256, shuffle=True)

class WrappedScheduler(torch.optim.lr_scheduler.ReduceLROnPlateau):
    def step(self, *args, **kwargs):
        raise Exception('Very step! Wow!')

class LeNet5(nn.Module):

    def __init__(self, n_classes):
        super(LeNet5, self).__init__()
        
        self.feature_extractor = nn.Sequential(            
            nn.Conv2d(in_channels=1, out_channels=6, kernel_size=5, stride=1),
            nn.Tanh(),
            nn.AvgPool2d(kernel_size=2),
            nn.Conv2d(in_channels=6, out_channels=16, kernel_size=5, stride=1),
            nn.Tanh(),
            nn.AvgPool2d(kernel_size=2),
            nn.Conv2d(in_channels=16, out_channels=120, kernel_size=5, stride=1, padding=1),
            nn.Tanh(),
            nn.AvgPool2d(kernel_size=2)
        )

        self.classifier = nn.Sequential(
            nn.Linear(in_features=120, out_features=84),
            nn.Tanh(),
            nn.Linear(in_features=84, out_features=n_classes),
        )


    def forward(self, x):
        x = self.feature_extractor(x)
        x = torch.flatten(x, 1)
        x = self.classifier(x)
        return x

model = LeNet5(n_classes=10)
optim = torch.optim.Adam(model.parameters())
scheduler = WrappedScheduler(optim)
criterion = nn.CrossEntropyLoss()

# this will throw an exception after one epoch.
runner = SupervisedRunner(model=model, device=torch.device('cpu'))
runner.train(model=model, 
             criterion=criterion, 
             optimizer=optim, 
             scheduler=scheduler, 
             loaders={"train": train_loader, "valid": test_loader}, verbose=True)

class SelfRunner(Runner):
    def _handle_batch(self, batch):
        x,y = batch

        p = self.model(x)

        loss = self.criterion(p, y)

        self.batch_metrics.update({'loss': loss.item()})

        if self.is_train_loader:
            loss.backward()
            self.optimizer.step()
            self.optimizer.zero_grad()

# this will not
runner = SelfRunner(device=torch.device('cpu'))
runner.train(model=model, 
             criterion=criterion, 
             optimizer=optim, 
             scheduler=scheduler, 
             loaders={"train": train_loader, "valid": test_loader}, verbose=True)
```

### Expected behavior
Scheduler should make a step after each epoch as in supervised runner.

### Environment
```bash
Catalyst version: 20.11
PyTorch version: 1.7.1+cu101
Is debug build: No
CUDA used to build PyTorch: 10.1
TensorFlow version: N/A
TensorBoard version: 2.3.0

OS: Ubuntu 18.04.2 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 9.1.85
GPU models and configuration: 
GPU 0: Quadro RTX 8000
GPU 1: Quadro RTX 8000
GPU 2: GeForce RTX 2080 Ti
GPU 3: GeForce RTX 2080 Ti

Nvidia driver version: 430.50
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] catalyst==20.11
[pip3] efficientnet-pytorch==0.6.3
[pip3] numpy==1.20.0
[pip3] numpydoc==1.1.0
[pip3] segmentation-models-pytorch==0.1.3
[pip3] tensorboard==2.3.0
[pip3] tensorboard-plugin-wit==1.7.0
[pip3] tensorboardX==2.1
[pip3] torch==1.7.1+cu101
[pip3] torchaudio==0.7.2
[pip3] torchvision==0.8.2+cu101
[conda] catalyst                  20.11                    pypi_0    pypi
[conda] efficientnet-pytorch      0.6.3                    pypi_0    pypi
[conda] mkl                       2020.4             h726a3e6_304    conda-forge
[conda] mkl-service               2.3.0            py37h8f50634_2    conda-forge
[conda] mkl_fft                   1.2.0            py37h161383b_1    conda-forge
[conda] mkl_random                1.2.0            py37h9fdb41a_1    conda-forge
[conda] numpy                     1.20.0           py37haa41c4c_0    conda-forge
[conda] numpy-base                1.18.5           py37h2f8d375_0  
[conda] numpydoc                  1.1.0                      py_1    conda-forge
[conda] segmentation-models-pytorch 0.1.3                    pypi_0    pypi
[conda] tensorboard               2.3.0                    pypi_0    pypi
[conda] tensorboard-plugin-wit    1.7.0                    pypi_0    pypi
[conda] tensorboardx              2.1                      pypi_0    pypi
[conda] torch                     1.7.1+cu101              pypi_0    pypi
[conda] torchaudio                0.7.2                    pypi_0    pypi
[conda] torchvision               0.8.2+cu101              pypi_0    pypi

```

### Checklist
- [x] bug description.
- [x] steps to reproduce.
- [x] expected behavior.
- [x] environment.
- [x] code sample / screenshots.
- [x] I know, that I could [join Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion.

**Comment by Scitator** [comment]
Hi,
yeah, that's correct behavior – Runner is more about pure custom pipelines, so there are no automatic callbacks included. Nevertheless, you could define them by yourself, here is an example - https://colab.research.google.com/drive/1O78hshITj_dfCNzT3BhrihXKcS_iZn4W?usp=sharing

PS. btw, we are working on a huge core redesign. in this case, this bug should be fixed in near future.

**Comment by MartSlaaf** [comment]
Hello @Scitator and thanks for your answer!

Then I have three questions:
1. Your link is not open-access. Is it as planned, should I require access?
2. Is there any way to get something close to the SupervisedRunner (that is, automatic callbacks and such a service), but only with my own `_handle_batch`? When I try it straight forward, I stumble upon some different behavior of `batch` and so on.
2. May it then be worthy to raise warnings when you pass the parameter which will not be processed? (which may be not worthy if you plan the redesign soon.)

**Comment by Scitator** [comment]
1. fixed link - https://colab.research.google.com/drive/1O78hshITj_dfCNzT3BhrihXKcS_iZn4W?usp=sharing
2. honestly speaking, just wait a couple of weeks please :)
- that's our next version 21xx demo - https://github.com/catalyst-team/catalyst/blob/kittylyst/examples/notebooks/demo%2021xx.ipynb
- and colab - https://colab.research.google.com/github/catalyst-team/catalyst/blob/kittylyst/examples/notebooks/demo%2021xx.ipynb (it's easier to navigate there thanks to the menu)
- and this version contains the fix you are looking for :)
3. sounds legit 👍 , it would be great if you could join our slack and share your feedback on the new demo - now it's a great time for improvement

**Comment by Scitator** [comment]
resolved in 21.xx versions ;)
